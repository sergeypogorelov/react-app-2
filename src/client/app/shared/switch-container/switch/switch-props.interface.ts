import { LabeledValue } from '../../../core/interfaces/common/labeled-value.interface';

export interface SwitchProps {
  left: LabeledValue;
  right: LabeledValue;
  currentValue: string;
  onChange: (newValue: string) => void;
}
