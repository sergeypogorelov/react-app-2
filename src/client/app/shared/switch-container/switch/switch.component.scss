@import '../../../../styles/variables';

.switch {
  &-item {
    &.active {
      background-color: $c-primary;
    }
    &-left {
      border-radius: 4px 0 0 4px;
    }
    &-right {
      border-radius: 0 4px 4px 0;
    }
    height: 35px;
    padding: 0 25px;
    font-size: 15px;
    color: $c-default;
    background-color: rgba(66, 66, 66, 0.68);
    border: none;
  }
}
