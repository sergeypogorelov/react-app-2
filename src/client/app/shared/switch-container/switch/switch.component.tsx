import './switch.component.scss';

import React from 'react';
import classNames from 'classnames';

import { SwitchProps } from './switch-props.interface';

export class Switch extends React.Component<SwitchProps> {
  render() {
    const { currentValue, left, right } = this.props;

    const activeClassNames = {
      left: classNames({
        'switch-item': true,
        'switch-item-left': true,
        active: left.value === currentValue,
      }),
      right: classNames({
        'switch-item': true,
        'switch-item-right': true,
        active: right.value === currentValue,
      }),
    };

    return (
      <div className="switch d-flex">
        <button
          className={activeClassNames.left}
          type="button"
          onClick={this.handleLeftButtonClick}
        >
          {left.label}
        </button>
        <button
          className={activeClassNames.right}
          type="button"
          onClick={this.handleRightButtonCallback}
        >
          {right.label}
        </button>
      </div>
    );
  }

  handleLeftButtonClick = () => {
    const { currentValue, left, onChange } = this.props;

    if (currentValue !== left.value) {
      onChange(left.value);
    }
  };

  handleRightButtonCallback = () => {
    const { currentValue, right, onChange } = this.props;

    if (currentValue !== right.value) {
      onChange(right.value);
    }
  };
}
