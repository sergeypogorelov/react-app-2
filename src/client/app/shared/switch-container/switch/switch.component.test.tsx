import React from 'react';
import renderer from 'react-test-renderer';

import { Switch } from './switch.component';

test('Switch component should render correctly', () => {
  const tree = renderer.create(<Switch />).toJSON();
  expect(tree).toMatchSnapshot();
});
