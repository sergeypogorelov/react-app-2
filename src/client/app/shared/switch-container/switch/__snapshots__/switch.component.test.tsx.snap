// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component should render correctly 1`] = `
<div
  className="switch d-flex"
>
  <button
    className="switch-item switch-item-left active"
  >
    Title
  </button>
  <button
    className="switch-item switch-item-right"
  >
    Genre
  </button>
</div>
`;
